### ISSUE TITLE
Various enhancements

### ISSUE THREAD

**Issue by gquerret** [Riverside-Software/sonar-openedge#1003]
On behalf of VIF

Two change requests in rules:
* Rule TooManyParameters would be easier to read if issues were raised on the callee instead of the caller
* Many false positives in ExpressionNoEffect with this pattern:

```
integer(xx) no-error. // Issue raised here. Same with date(xx) no-error.
if error-status:error then do:
  // handle invalid input
end.
```

**Comment by gquerret** [comment]
Second bullet point fixed in the develop branch

**Comment by gquerret** [comment]
Deprecate TooManyParameters rule, new one reports issues on methods / procedures / functions
